tensorflow/core/kernels/random_op_gpu.cu.cc

```diff
@@ -141,7 +141,7 @@ struct FillPhiloxRandomKernel<Distribution, false> {
       const typename Distribution::ResultType samples = dist(&gen);
       copier(&data[offset], samples);
 
-      offset += (total_thread_count - 1) * kGroupSize;
+      offset += total_thread_count * kGroupSize;
       gen.Skip(total_thread_count - 1);
```

Comment:
Hmm...you've verified that the 0th thread on the 2nd pass through the loop doesn't generate the same result as the last thread on the first pass?  It seems like that must be the case.  I'd like to understand how that's possible.